pkg/datapath/loader/compile.go

```diff
@@ -163,18 +163,17 @@ func GetBPFCPU() string {
 			// added in the same release.
 			// We want to enable v3 only on kernels 5.10+ where we have
 			// tested it and need it to work around complexity issues.
-			if h := manager.GetHelpers("sched_cls"); h != nil {
+			h := manager.GetHelpers("sched_cls")
+			if manager.GetMisc().HaveV3ISAExtension && h != nil {
```

Comment:
Out of curiosity, was there a case where h != nil  but `HaveV3ISAExtension` was false?

Reply:
Not unless there are some weird backports being performed e.g. on RHEL. I just wanted to make the condition explicit :-)